analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextBaseTest.ShouldAnalyzeTree.cs

```diff
@@ -331,58 +332,62 @@ public class GenericAttribute<T> : Attribute { }
     [DataRow("Foo", new string[] { "Foo" }, ProjectType.Test, true)]
     [DataRow("Foo", new string[] { "NotFoo" }, ProjectType.Test, true)]
-    public void ShouldAnalyzeTree_Inclusions_ReturnExpected(string filePath, string[] inclusions, ProjectType projectType, bool shouldAnalyze) =>
-        AssertShouldAnalyzeTree_SonarLint(filePath, projectType, shouldAnalyze, inclusions: inclusions);
+    public void ShouldAnalyzeTree_Inclusions_ReturnExpected(string filePath, string[] inclusions, ProjectType projectType, bool expectedResult) =>
+        ShouldAnalyzeTree_WithExclusionInclusionParametersSet_ReturnsTrueForIncludedFilesOnly(filePath, projectType, expectedResult, inclusions: inclusions);
 
     [DataTestMethod]
     [DataRow("Foo", new string[] { "Foo" }, ProjectType.Product, true)]
     [DataRow("Foo", new string[] { "NotFoo" }, ProjectType.Product, true)]
     [DataRow("Foo", new string[] { "Foo" }, ProjectType.Test, true)]
     [DataRow("Foo", new string[] { "NotFoo" }, ProjectType.Test, false)]
-    public void ShouldAnalyzeTree_TestInclusions_ReturnExpected(string filePath, string[] testInclusions, ProjectType projectType, bool shouldAnalyze) =>
-        AssertShouldAnalyzeTree_SonarLint(filePath, projectType, shouldAnalyze, testInclusions: testInclusions);
+    public void ShouldAnalyzeTree_TestInclusions_ReturnExpected(string filePath, string[] testInclusions, ProjectType projectType, bool expectedResult) =>
+        ShouldAnalyzeTree_WithExclusionInclusionParametersSet_ReturnsTrueForIncludedFilesOnly(filePath, projectType, expectedResult, testInclusions: testInclusions);
 
     [DataTestMethod]
     [DataRow("Foo", new string[] { "Foo" }, new string[] { "Foo" }, ProjectType.Product, false)]
     [DataRow("Foo", new string[] { "NotFoo" }, new string[] { "Foo" }, ProjectType.Product, false)]
     [DataRow("Foo", new string[] { "Foo" }, new string[] { "NotFoo" }, ProjectType.Product, true)]
     [DataRow("Foo", new string[] { "NotFoo" }, new string[] { "NotFoo" }, ProjectType.Product, false)]
-    public void ShouldAnalyzeTree_MixedInput_ReturnExpected(string filePath, string[] inclusions, string[] exclusions, ProjectType projectType, bool shouldAnalyze) =>
-        AssertShouldAnalyzeTree_SonarLint(filePath, projectType, shouldAnalyze, inclusions: inclusions, exclusions: exclusions);
-
-    private void AssertShouldAnalyzeTree_SonarLint(
+    [DataRow("Foo", new string[] { "Foo" }, new string[] { "Foo" }, ProjectType.Test, true)]
+    [DataRow("Foo", new string[] { "NotFoo" }, new string[] { "Foo" }, ProjectType.Test, true)]
+    [DataRow("Foo", new string[] { "Foo" }, new string[] { "NotFoo" }, ProjectType.Test, true)]
+    [DataRow("Foo", new string[] { "NotFoo" }, new string[] { "NotFoo" }, ProjectType.Test, true)]
+    public void ShouldAnalyzeTree_MixedInput_ReturnExpected(string filePath, string[] inclusions, string[] exclusions, ProjectType projectType, bool expectedResult) =>
```

Comment:
For the test project type the test  exclusion inclusion parameters should be set, so actually, some exclusion happens.